### PR TITLE
feat: commands rbac

### PR DESCRIPTION
**Pull request by hamidzr**
## Commentary (optional)

we'll separately want to add:
- cli support
- ee tests

DET-8817

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/internal/api_command.go [hunk 1]
```diff
@@ -241,7 +246,25 @@ func (a *apiServer) GetCommand(
...
+	}
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	err = command.AuthZProvider.Get().CanTerminateNSC(
+		ctx, *curUser, model.AccessScopeID(targetCmd.Command.WorkspaceId),
+	)
+	err = apiutils.MapAndFilterErrors(err, nil, nil)
```
Comment (hamidzr) [inline master/internal/api_command.go:261]: this is kind of a common pattern and we might want to add more logic here about sometimes reporting perm errors as 404 and sometimes expose the  permission issue.

Comment (eecsliu) [reply master/internal/api_command.go:261]: Wouldn't it be 404 errors for listing commands but permission issue for other things? Maybe it's worth having a conversation about, but I'm not sure it returning 404 matters much if they don't have permission to view it anyways?

File: master/internal/api_command.go [hunk 2]
```diff
@@ -251,7 +274,27 @@ func (a *apiServer) KillCommand(
...
+	}
+
+	err = command.AuthZProvider.Get().CanSetNSCsPriority(
+		ctx, *curUser, model.AccessScopeID(targetCmd.Command.WorkspaceId), int(req.Priority),
+	)
+	err = apiutils.MapAndFilterErrors(err, nil, nil)
+	switch err.(type) {
+	case nil:
+		// do nothing
+	case authz.PermissionDeniedError:
```
Comment (eecsliu) [inline master/internal/api_command.go:293]: ditto above

File: master/internal/api_ntsc_intg_test.go [hunk 3]
```diff
@@ -77,92 +95,139 @@ func TestTasksCountAuthZ(t *testing.T) {
...
 	api, authz, curUser, ctx := setupNTSCAuthzTest(t)
+	var err error
 
-	nbsActor := actor.AddrFromString("/notebooks")
-	nbID := setupMockNBActor(t, api.m)
 	// check permission errors are returned with not found status and follow the same pattern.
 	authz.On("CanGetNSC", mock.Anything, curUser, mock.Anything, mock.Anything).Return(
 		false, nil,
-	).Once()
+	).Times(2)
```
Comment (eecsliu) [inline master/internal/api_ntsc_intg_test.go:103]: question: whats the times 2 for?

**Review by eecsliu — COMMENTED** [review]
Stuff looks good, I think we should discuss more about the 404 errors tho

File: master/internal/api_ntsc_intg_test.go (hunk 3 above)
Comment (hamidzr) [reply master/internal/api_ntsc_intg_test.go:103]: asserts that it can only be called up to two times

File: master/internal/api_command.go (hunk 1 above)
Comment (hamidzr) [reply master/internal/api_command.go]: we chatted about this offiline. ref https://determined-ai.slack.com/archives/C03LE5Z8R46/p1672776613155459?thread_ts=1672772755.830529&cid=C03LE5Z8R46

File: master/internal/api_command.go [hunk 4]
```diff
@@ -188,29 +190,44 @@ func (a *apiServer) getCommandLaunchParams(ctx context.Context, req *protoComman
...
-		return ok
-	})
+	limitedScopes, err := command.AuthZProvider.Get().AccessibleScopes(
+		ctx, *curUser, model.AccessScopeID(req.WorkspaceId),
+	)
 	if err != nil {
 		return nil, err
 	}
+	if req.WorkspaceId != 0 && len(limitedScopes) == 0 {
+		// report missing permissions as a 404 on explicit
```
Comment (eecsliu) [inline master/internal/api_command.go:223]: nit: I feel like this comment is extraneous because I can understand it by reading line 222.

**Review by eecsliu — APPROVED** [review]
Nice! 🚢

File: master/internal/api_command.go (hunk 4 above)
Comment (hamidzr) [reply master/internal/api_command.go:223]: 👍🏼  lets take it out in another pr (to avoid another ci round)

**Comment by hamidzr** [comment]
thanks for the review

**Pull request merged**